OS Upgrade parameter addition

roshnipatel999 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Added support for Windows Server OS upgrade while migrating the server to Azure using Azure Migrate. -->

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

ankit-gautam23 reviewed (approved):
LGTM

singhabh27 commented on src/Migrate/custom/Start-AzMigrateServerMigration.ps1:
```diff
@@ -132,10 +139,22 @@ function Start-AzMigrateServerMigration {
+                if ($null -eq $SupportedOsVersions) {
+                    throw "There is no supported target OS available. Please check or remove the OsUpgradeVersion input." 
+                }
+                elseif ($SupportedOsVersions -contains $OsUpgradeVersion) {
```
Why contains? It should be -eq

roshnipatel999 commented:
> @roshnipatel999 Please provide a description statement for this change
@msJinLei I have added the description and did a minor change as well. Please review

roshnipatel999 commented:
@msJinLei Please review

msJinLei commented:
> roshnipatel999

Sorry, I mean we need a change log for your when it is release. Can you provide it?
Refer to this link
https://github.com/Azure/azure-powershell/blob/main/src/RecoveryServices/RecoveryServices/ChangeLog.md

roshnipatel999 commented:
@msJinLei I have modified the description.

msJinLei commented:
> @msJinLei I have modified the description.

where do you put your description?

roshnipatel999 commented:
@msJinLei I edited the PR description. In case not visible this is what I added:
Added support for Windows Server OS upgrade while migrating the server to Azure using Azure Migrate
